components/Post.php

```diff
@@ -93,7 +93,24 @@ protected function loadPost()
             $post = $post->isPublished();
         }
 
-        $post = $post->first();
+        try {
+            $post = $post->firstOrFail();
```

Comment:
@maartenmachiels Remove everything below this line in the `try` block, as the same code is specified underneath the `catch` block.